### ISSUE TITLE
Enhancement - Futur EPWING support and Kenkyusha

### ISSUE THREAD

**Issue by epistularum** [Garethp/RikaiRebuilt#7]
I posted this already the nazeka github and the yomichan github but since there are so many forks I figured I could post it here aswell : 

> Something I've always wished from any pop up dictionary is a good support for the kenkyusha waei dictionary.
> The best support at the moment is still rikaisama and a lot of people stick to firefox 57/waterfox/palemoon... for the sole purpose of rikaisama and its regex deletion.
> 
> It's a good occasion to improve on rikaisama's EPWING feature (especially for kenkyusha due to its large number of of examples and what not). 
> 
> For the moment with rikaisama and its regex feature we can clean up the definition field BUT in this process we're forced to delete all the examples.
> I've been trying to find a fix using regex but I'm not competent enough make a breakthrough, here's what I figured out so far.
> 
> Example of an entry :
> 
> ```
> まにあう【間に合う】 ﾛｰﾏ(maniau)
> 1 〔時間に遅れない〕 be in time 《for…》.
> ▲7 時の列車に間に合う　catch [make] the 7 o'clock train
> ・締め切りに間に合う　meet the deadline
> ・開演に間に合う　arrive before curtain time
> ▲9 時の札幌行きに間に合うように空港に着いた.　I arrived in time for the nine o'clock flight to Sapporo.
> ・「間に合うかな」「走っても間に合いそうにないね」　"Will we be in time?"―"It doesn't look like we'll be in time even if we run."
> 2 〔役に立つ〕 answer [serve, suit, meet] the purpose; be useful; be serviceable; be of ￼use [service]; be good enough; 〔十分である〕 be enough; 〔用意ができる〕 be ready; 〔必要をみたす〕 meet the requirements; serve ￼the [one's] turn [need].
> ▲「費用はどのぐらいかな」「5 万もあれば間に合うよ」　"And what is the expense?"―"Fifty-thousand yen should cover it."
> ・これだけあれば丸 1 年は間に合う.　This will ￼last us [see us through] one whole year. ｜ This will be enough for a whole year.
> ```
> 
> Where all entries starting with "▲" or "・" are examples and all entries matching this regex are definitions :
> 
> Regular expression that matches everything that is not a definition :
> `\n[^″*〖〈《⇒＝➡【〔(〜A-Za-z0-9].*`
> 
> Regular expression that matches definitions+one line below :
> `\n[″*〖〈《⇒＝➡【〔(〜A-Za-z0-9].*\n.*`
> 
> The perfect result should look like this(keeping one example for each definition) :
> ```
> まにあう【間に合う】 ﾛｰﾏ(maniau)
> 1 〔時間に遅れない〕 be in time 《for…》.
> ▲7 時の列車に間に合う　catch [make] the 7 o'clock train
> 2 〔役に立つ〕 answer [serve, suit, meet] the purpose; be useful; be serviceable; be of ￼use [service]; be good enough; 〔十分である〕 be enough; 〔用意ができる〕 be ready; 〔必要をみたす〕 meet the requirements; serve ￼the [one's] turn [need].
> ▲「費用はどのぐらいかな」「5 万もあれば間に合うよ」　"And what is the expense?"―"Fifty-thousand yen should cover it."
> ```
> tldr : adding support for kenkyusha while keeping only 1 example for each definition would be a godsend for anyone learning Japanese and will finally achieve a breakthrough in the dictionary pop up app nonsense.

**Comment by Garethp** [comment]
Oh wow, that's really interesting. Do you know of any other EPWING dictionaries that do something like this?

Can you send me the Kenkyusha dictionary, or a link where I can grab it? Feel free to email me at blackfireweb@gmail.com

**Comment by epistularum** [comment]
> Oh wow, that's really interesting. Do you know of any other EPWING dictionaries that do something like this?
> 
> Can you send me the Kenkyusha dictionary, or a link where I can grab it? Feel free to email me at [blackfireweb@gmail.com](mailto:blackfireweb@gmail.com)

I'm not sure to understand what you mean by "do something like this".
I have sent you an email.
